include/IECore/MeshAlgo.h

```diff
@@ -54,6 +54,8 @@ std::pair<PrimitiveVariable, PrimitiveVariable> calculateTangents( const MeshPri
 	const std::string &position = "P"
 );
 
+PrimitiveVariable resamplePrimitiveVariable( const MeshPrimitive *mesh, const std::string &primitiveVariable, PrimitiveVariable::Interpolation interpolation );
```

Comment:
You've convinced me on the const PrimitiveVariable& API. Also I don't know why I used const MeshPrimitive* rather than const MeshPrimitive&.  What to do about MeshTangents named position primitive variable? I'd like to leave it until the Cortex 10 UV change if that sounds reasonable?

Regarding the copy I don't like any of the options I've come up with:
1) new bool to define the always copy behaviour, comment and duplicate of the data if the always copy bool parameter is set.
2) return primvar with empty data if interpolation matches & comment 
3) Just add a  comment to explain what's going on.

I suppose i would choose (2) as then the caller has the option of doing what they want with no additional complication to the API.